cmd/fs-v1-multipart.go

```diff
@@ -345,126 +211,58 @@ func (fs fsObjects) ListMultipartUploads(bucket, object, keyMarker, uploadIDMark
 	return result, nil
 }
 
-// newMultipartUpload - wrapper for initializing a new multipart
-// request, returns back a unique upload id.
-//
-// Internally this function creates 'uploads.json' associated for the
-// incoming object at '.minio.sys/multipart/bucket/object/uploads.json' on
-// all the disks. `uploads.json` carries metadata regarding on going
-// multipart operation on the object.
-func (fs fsObjects) newMultipartUpload(bucket string, object string, meta map[string]string) (uploadID string, err error) {
-	// Initialize `fs.json` values.
-	fsMeta := newFSMetaV1()
-
-	// Save additional metadata.
-	fsMeta.Meta = meta
-
-	uploadID = mustGetUUID()
-	initiated := UTCNow()
-
-	// Add upload ID to uploads.json
-	uploadsPath := pathJoin(bucket, object, uploadsJSONFile)
-	rwlk, err := fs.rwPool.Create(pathJoin(fs.fsPath, minioMetaMultipartBucket, uploadsPath))
-	if err != nil {
-		return "", toObjectErr(errors.Trace(err), bucket, object)
-	}
-	defer rwlk.Close()
-
-	uploadIDPath := pathJoin(bucket, object, uploadID)
-	fsMetaPath := pathJoin(fs.fsPath, minioMetaMultipartBucket, uploadIDPath, fsMetaJSONFile)
-	metaFile, err := fs.rwPool.Create(fsMetaPath)
-	if err != nil {
-		return "", toObjectErr(errors.Trace(err), bucket, object)
-	}
-	defer metaFile.Close()
-
-	// Add a new upload id.
-	if err = fs.addUploadID(bucket, object, uploadID, initiated, rwlk); err != nil {
-		return "", toObjectErr(err, bucket, object)
-	}
-
-	// Write all the set metadata.
-	if _, err = fsMeta.WriteTo(metaFile); err != nil {
-		return "", toObjectErr(err, bucket, object)
-	}
-
-	// Return success.
-	return uploadID, nil
-}
-
 // NewMultipartUpload - initialize a new multipart upload, returns a
 // unique id. The unique id returned here is of UUID form, for each
 // subsequent request each UUID is unique.
 //
 // Implements S3 compatible initiate multipart API.
-func (fs fsObjects) NewMultipartUpload(bucket, object string, meta map[string]string) (string, error) {
+func (fs *fsObjects) NewMultipartUpload(bucket, object string, meta map[string]string) (string, error) {
 	if err := checkNewMultipartArgs(bucket, object, fs); err != nil {
-		return "", err
+		return "", toObjectErr(errors.Trace(err), bucket)
 	}
 
 	if _, err := fs.statBucketDir(bucket); err != nil {
-		return "", toObjectErr(err, bucket)
+		return "", toObjectErr(errors.Trace(err), bucket)
```

Comment:
You could avoid double tracing here err returned by checkNewMultipartArgs is already traced.